staging/src/k8s.io/apiserver/pkg/server/secure_serving.go

```diff
@@ -195,6 +195,15 @@ func (s *SecureServingInfo) Serve(handler http.Handler, shutdownTimeout time.Dur
 			http2Options.MaxConcurrentStreams = 100
 		}
 
+		// use the overridden HTTP2ReadIdleTimeout value, without HTTP2ReadIdleTimeout HTTP2PingTimeout has no usage
+		if s.HTTP2ReadIdleTimeout > 0 {
```

Comment:
Also, I can remove this if and just pass the actual value, whether it's zero or not if you wish.